code/modules/mining/machine_processing.dm

```diff
@@ -116,10 +116,7 @@
 
 		if("toggle")
 			processing_machine.on = !processing_machine.on
-                     if(on)
-			  processing_machine.begin_processing()
```

Comment:
I didn't type the correct indentation , cause it was on my phone. I expect you to fix that and then implement my changes not just give up and go back

Reply:
Good point I didn't check the process proc. So you don't need to explicitly end processing here but you should only begin processing if on is TRUE.

Also inside your process proc replace end processing with return PROCESS_KILL and then your good